addons/binding/org.openhab.binding.chamberlainmyq/src/main/java/org/openhab/binding/chamberlainmyq/handler/ChamberlainMyQHandler.java

```diff
@@ -0,0 +1,176 @@
+     */
+    protected ChamberlainMyQDeviceConfig deviceConfig;
+
+    private Logger logger = LoggerFactory.getLogger(ChamberlainMyQHandler.class);
+
+    /**
+     * Creates a new instance of this class for the {@link Thing}.
+     *
+     * @param thing the thing that should be handled, not null.
+     */
+    public ChamberlainMyQHandler(Thing thing) {
+        super(thing);
+
+        String id = getThing().getProperties().get(MYQ_ID);
+        logger.debug("Thing ID: {}", id);
+        this.deviceConfig = new ChamberlainMyQDeviceConfig(getThing().getProperties());
+        // parseConfig(config); */
+        logger.info("Initializing a MyQ device: \n{}", deviceConfig.asString());
+    }
+
+    /**
+     * Parse the configuration of this thing.
+     *
+     * @param jsonConfigString the string containing the configuration of this thing as returned by the hub (in JSON).
+     */
+    protected void parseConfig(String jsonConfigString) {
+        logger.debug("Parsing a thing's config: {}", jsonConfigString);
+        JsonParser parser = new JsonParser();
+        deviceConfig.readConfigFromJson(parser.parse(jsonConfigString).getAsJsonObject());
+    }
+
+    /**
+     * Used to retrieve the {@link ChamberlainMyQGatewayHandler} controlling this thing.
+     *
+     * @return this thing gateway handler, null if it hasn't been set yet.
+     */
+    protected ChamberlainMyQGatewayHandler getGatewayHandler() {
+        Bridge gateway = getBridge();
+        return gateway == null ? null : (ChamberlainMyQGatewayHandler) gateway.getHandler();
+    }
+
+    /**
+     * Function called when a communication error between the gateway and the thing has been detected.
+     */
+    protected void handleCommunicationError(String errorMessage) {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, errorMessage);
+    }
+
+    private abstract class ChamberlainMyQDeviceRequestCallback implements RequestCallback {
+        /**
+         * Handler of the thing that should for which the configuration should be set.
+         */
+        protected ChamberlainMyQHandler handler;
+
+        /**
+         * Creates a new instance of this class.
+         *
+         * @param handler The handler for which the configuration should be read.
+         */
+        public ChamberlainMyQDeviceRequestCallback(ChamberlainMyQHandler handler) {
+            Preconditions.checkArgument(handler != null, "The argument |handler| must not be null.");
+            this.handler = handler;
+        }
+
+        @Override
+        public void onError(String error) {
+            handler.handleCommunicationError(error);
+        }
+    }
+
+    //////////// Read State functions ////////////
+
+    /**
+     * Specialization of a {link RequestCallback} to read a device configuration.
+     *
+     * @author Scott Hanson
+     */
+    private class ReadDeviceStateCallback extends ChamberlainMyQDeviceRequestCallback {
```

Comment:
make the class "static" otherwise the surrounding object will leak in.

Reply:
I don't understand what you mean?

Reply:
`private static class ...`
https://www.tutorialspoint.com/What-is-a-static-class-in-Java